src/vs/workbench/contrib/terminal/common/terminalContextKey.ts

```diff
@@ -39,6 +39,7 @@ export const enum TerminalContextKeyStrings {
 	ShellType = 'terminalShellType',
 	InTerminalRunCommandPicker = 'inTerminalRunCommandPicker',
 	TerminalShellIntegrationEnabled = 'terminalShellIntegrationEnabled',
+	TerminalHasCwdDetectionCapability = 'terminalHasNaiveCwdDetectionCapability',
```

Comment:
This is somewhat user facing, so how about something closer to `terminalHasCwdDetection` or `terminalShellIntegrationCwdDetection`?